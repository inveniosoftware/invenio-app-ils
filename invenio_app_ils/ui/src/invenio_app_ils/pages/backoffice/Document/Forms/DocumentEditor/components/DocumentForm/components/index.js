export { TagsField } from './TagsField';
export { AlternativeAbstracts } from './AlternativeAbstracts';
export { TableOfContent } from './TableOfContent';
export { UrlsField } from './UrlsField';
