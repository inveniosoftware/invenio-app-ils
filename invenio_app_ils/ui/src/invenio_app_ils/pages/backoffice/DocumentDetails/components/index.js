export { DocumentDetails } from './DocumentDetails';
export { DocumentMetadata } from './DocumentMetadata';
export { DocumentPendingLoans } from './DocumentPendingLoans';
export { DocumentAccessRestrictions } from './DocumentAccessRestrictions';
