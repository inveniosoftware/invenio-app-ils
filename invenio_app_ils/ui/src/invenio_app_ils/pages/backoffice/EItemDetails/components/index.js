export { EItemDetails } from './EItemDetails';
export { EItemMetadata } from './EItemMetadata';
