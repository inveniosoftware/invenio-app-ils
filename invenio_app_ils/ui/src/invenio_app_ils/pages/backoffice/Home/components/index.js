export { LoansCard } from './widgets/LoansCard';
export { DocumentsCard } from './widgets/DocumentsCard';
export { ACQRequestsCard } from './widgets/ACQRequestsCard';
export { ILLCard } from './widgets/ILLCard';
export { OverbookedDocumentsList } from './lists/OverbookedDocumentsList';
export { OverdueLoansList } from './lists/OverdueLoansList';
export { IdleLoansList } from './lists/IdleLoansList';
export { RenewedLoansList } from './lists/RenewedLoansList';
