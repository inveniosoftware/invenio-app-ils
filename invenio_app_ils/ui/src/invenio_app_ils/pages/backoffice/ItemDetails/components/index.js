export { ItemDetails } from './ItemDetails';
export { ItemMetadata } from './ItemMetadata';
export { ItemPastLoans } from './ItemPastLoans';
