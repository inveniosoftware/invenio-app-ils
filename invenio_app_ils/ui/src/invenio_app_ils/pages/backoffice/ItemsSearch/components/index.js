export { ResultsList } from './ResultsList';
export { SearchBar } from './SearchBar';
