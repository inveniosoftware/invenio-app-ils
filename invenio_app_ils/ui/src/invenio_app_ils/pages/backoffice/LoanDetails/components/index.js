export { LoanActions } from './LoanActions';
export { LoanDetails } from './LoanDetails';
export { LoanMetadata } from './LoanMetadata';
export { AvailableItems } from './AvailableItems';
