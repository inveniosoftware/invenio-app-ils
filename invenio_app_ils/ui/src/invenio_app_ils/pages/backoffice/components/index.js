export { Header } from './Header';
export { MetadataTable } from './MetadataTable';
export { DeleteRecordModal } from './DeleteRecordModal';
export {
  ExportReactSearchKitResults,
  ExportSearchResults,
} from './ExportSearchResults';
export { OverdueLoanSendMailModal } from './OverdueLoanSendMailModal';
export { Sidebar } from './Sidebar';
export { DocumentList } from './DocumentList';
export { ItemListEntry } from './ItemListEntry';
