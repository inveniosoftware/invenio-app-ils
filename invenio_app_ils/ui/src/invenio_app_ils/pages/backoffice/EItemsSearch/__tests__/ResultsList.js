import React from 'react';
import omit from 'lodash/omit';
import { mount } from 'enzyme';
import { Settings } from 'luxon';
import { fromISO, toISO } from '../../../../common/api/date';
import { ResultsList } from '../components';
import { formatter } from '../../../../common/components/ResultsTable/formatters';

Settings.defaultZoneName = 'utc';

describe('EItemsSearch ResultsList tests', () => {
  const stringDate = fromISO('2018-01-01T11:05:00+01:00');

  const results = [
    {
      id: 987,
      created: toISO(stringDate),
      updated: toISO(stringDate),
      metadata: {
        eitem_pid: 987,
        description: 'A description.',
        document_pid: 1342,
        document: {
          document_pid: 1342,
        },
        internal_notes: 'Internal notes.',
      },
    },
  ];

  let component;
  afterEach(() => {
    component.unmount();
  });

  it('should not render when empty results', () => {
    component = mount(
      <ResultsList results={[]} viewDetailsClickHandler={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a list of results', () => {
    component = mount(
      <ResultsList results={results} viewDetailsClickHandler={() => {}} />
    );
    expect(component).toMatchSnapshot();
    const firstResult = results[0];
    const resultRows = component
      .find('TableRow')
      .filterWhere(
        element => element.prop('data-test') === firstResult.metadata.eitem_pid
      );
    expect(resultRows).toHaveLength(1);

    const mappedStatusElements = resultRows
      .find('TableCell')
      .filterWhere(
        element =>
          element.prop('data-test') ===
          'Description-' + firstResult.metadata.eitem_pid
      );
    expect(mappedStatusElements).toHaveLength(1);

    expect(mappedStatusElements.text()).toEqual(
      firstResult.metadata.description
    );
  });

  it('should call click handler on view details click', () => {
    const mockedClickHandler = jest.fn();
    component = mount(
      <ResultsList
        results={results}
        viewDetailsClickHandler={mockedClickHandler}
      />
    );
    const firstId = results[0].metadata.eitem_pid;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('button');
    button.simulate('click');
    const expected = omit(formatter.eitem.toTable(results[0]), [
      'Created',
      'Updated',
    ]);
    expect(mockedClickHandler).toHaveBeenCalledWith(expected);
  });
});
