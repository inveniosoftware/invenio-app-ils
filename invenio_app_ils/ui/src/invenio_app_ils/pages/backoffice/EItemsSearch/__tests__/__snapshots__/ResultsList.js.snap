// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EItemsSearch ResultsList tests should not render when empty results 1`] = `
<ResultsList
  results={Array []}
  viewDetailsClickHandler={[Function]}
/>
`;

exports[`EItemsSearch ResultsList tests should render a list of results 1`] = `
<ResultsList
  results={
    Array [
      Object {
        "created": "2018-01-01T10:05:00.000Z",
        "id": 987,
        "metadata": Object {
          "description": "A description.",
          "document": Object {
            "document_pid": 1342,
          },
          "document_pid": 1342,
          "eitem_pid": 987,
          "internal_notes": "Internal notes.",
        },
        "updated": "2018-01-01T10:05:00.000Z",
      },
    ]
  }
  viewDetailsClickHandler={[Function]}
>
  <ResultsTable
    fixed={false}
    headerActionClickHandler={null}
    headerActionComponent={
      <NewButton
        clickHandler={[Function]}
        disabled={false}
        fluid={false}
        text="New eitem"
      />
    }
    name="eitems"
    rowActionClickHandler={[Function]}
    rows={
      Array [
        Object {
          "Description": "A description.",
          "Document ID": 1342,
          "ID": 987,
          "Internal Notes": "Internal notes.",
          "Open access": "No",
        },
      ]
    }
    seeAllComponent={null}
    showMaxRows={10}
    singleLine={false}
    subtitle=""
    title=""
  >
    <Segment>
      <div
        className="ui segment"
      >
        <Grid>
          <div
            className="ui grid"
          >
            <GridRow>
              <div
                className="row"
              >
                <GridColumn
                  verticalAlign="middle"
                  width={13}
                >
                  <div
                    className="middle aligned thirteen wide column"
                  />
                </GridColumn>
                <GridColumn
                  textAlign="right"
                  width={3}
                >
                  <div
                    className="right aligned three wide column"
                  >
                    <NewButton
                      clickHandler={[Function]}
                      disabled={false}
                      fluid={false}
                      text="New eitem"
                    >
                      <Button
                        as="button"
                        icon={true}
                        labelPosition="left"
                        onClick={[Function]}
                        positive={true}
                        size="small"
                      >
                        <button
                          className="ui small icon positive left labeled button"
                          onClick={[Function]}
                        >
                          <Icon
                            as="i"
                            name="plus"
                          >
                            <i
                              aria-hidden="true"
                              className="plus icon"
                            />
                          </Icon>
                          New eitem
                        </button>
                      </Button>
                    </NewButton>
                  </div>
                </GridColumn>
              </div>
            </GridRow>
          </div>
        </Grid>
        <Table
          as="table"
          selectable={true}
          striped={true}
        >
          <table
            className="ui selectable striped table"
          >
            <ResultsTableHeader
              columns={
                Array [
                  "ID",
                  "Document ID",
                  "Open access",
                  "Description",
                  "Internal Notes",
                ]
              }
              withRowAction={true}
            >
              <TableHeader
                as="thead"
              >
                <thead
                  className=""
                >
                  <TableRow
                    as="tr"
                    cellAs="td"
                    data-test="header"
                  >
                    <tr
                      className=""
                      data-test="header"
                    >
                      <TableHeaderCell
                        as="th"
                        collapsing={true}
                        width={2}
                      >
                        <TableCell
                          as="th"
                          className=""
                          collapsing={true}
                          width={2}
                        >
                          <th
                            className="collapsing two wide"
                          />
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="ID"
                        width={2}
                      >
                        <TableCell
                          as="th"
                          className=""
                          width={2}
                        >
                          <th
                            className="two wide"
                          >
                            ID
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Document ID"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Document ID
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Open access"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Open access
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Description"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Description
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Internal Notes"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Internal Notes
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                    </tr>
                  </TableRow>
                </thead>
              </TableHeader>
            </ResultsTableHeader>
            <ResultsTableBody
              columns={
                Array [
                  "ID",
                  "Document ID",
                  "Open access",
                  "Description",
                  "Internal Notes",
                ]
              }
              rowActionClickHandler={[Function]}
              rows={
                Array [
                  Object {
                    "Description": "A description.",
                    "Document ID": 1342,
                    "ID": 987,
                    "Internal Notes": "Internal notes.",
                    "Open access": "No",
                  },
                ]
              }
            >
              <TableBody
                as="tbody"
              >
                <tbody
                  className=""
                >
                  <TableRow
                    as="tr"
                    cellAs="td"
                    data-test={987}
                    key="987"
                  >
                    <tr
                      className=""
                      data-test={987}
                    >
                      <TableCell
                        as="td"
                        textAlign="center"
                      >
                        <td
                          className="center aligned"
                        >
                          <Button
                            as="button"
                            circular={true}
                            compact={true}
                            data-test="btn-view-details-987"
                            icon="eye"
                            onClick={[Function]}
                          >
                            <button
                              className="ui circular compact icon button"
                              data-test="btn-view-details-987"
                              onClick={[Function]}
                            >
                              <Icon
                                as="i"
                                name="eye"
                              >
                                <i
                                  aria-hidden="true"
                                  className="eye icon"
                                />
                              </Icon>
                            </button>
                          </Button>
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="ID-987"
                        key="0-987"
                      >
                        <td
                          className=""
                          data-test="ID-987"
                        >
                          987
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Document ID-987"
                        key="1-987"
                      >
                        <td
                          className=""
                          data-test="Document ID-987"
                        >
                          1342
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Open access-987"
                        key="2-987"
                      >
                        <td
                          className=""
                          data-test="Open access-987"
                        >
                          No
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Description-987"
                        key="3-987"
                      >
                        <td
                          className=""
                          data-test="Description-987"
                        >
                          A description.
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Internal Notes-987"
                        key="4-987"
                      >
                        <td
                          className=""
                          data-test="Internal Notes-987"
                        >
                          Internal notes.
                        </td>
                      </TableCell>
                    </tr>
                  </TableRow>
                </tbody>
              </TableBody>
            </ResultsTableBody>
            <ResultsTableFooter
              allRowsNumber={1}
              columnsNumber={5}
              seeAllComponent={null}
              showMaxRows={10}
            />
          </table>
        </Table>
      </div>
    </Segment>
  </ResultsTable>
</ResultsList>
`;
